### PR TITLE
Simplify DA-BEAT projects linking + reference improvements 

### PR DESCRIPTION
**Pull request by sdlyy**
Resolves L2B-6432

### CODE REVIEW

File: packages/config/src/projects/other/da-beat/types/DaBridge.ts [hunk 1]
```diff
@@ -115,9 +116,9 @@ type CommonDaBridge = {
   }
 
   /**
-   * List of projects given bridge is being used in
+   * List of projects' references given bridge is being used in
    */
-  usedIn: ProjectId[]
+  usedIn: DaProjectReference[]
```
Comment (torztomasz) [inline packages/config/src/projects/other/da-beat/types/DaBridge.ts:121]: I kind a don't like the word reference here, especially that usedIn is the key.

File: packages/config/src/projects/other/da-beat/utils/link-by-da.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const ignoreArchivedFilter = (layer: Layer2 | Layer3) => !layer.isArchived
+
+  const l2s = toDaProjectReference(
+    layer2s.filter(filterFn).filter(ignoreArchivedFilter),
+  )
+  const l3s = toDaProjectReference(
+    layer3s.filter(filterFn).filter(ignoreArchivedFilter),
+  )
+
+  return [...l2s, ...l3s]
```
Comment (torztomasz) [inline packages/config/src/projects/other/da-beat/utils/link-by-da.ts:29]:
```suggestion
  const usedProjects = [...layer2s, ...layer3s].filter(filterFn).filter((layer) => !layer.isArchived)

  return toDaProjectReference(usedProjects)
```

**Review by torztomasz — CHANGES_REQUESTED** [review]

File: packages/config/src/projects/other/da-beat/utils/link-by-da.ts (hunk 2 above)
Comment (sdlyy) [reply packages/config/src/projects/other/da-beat/utils/link-by-da.ts]: https://github.com/l2beat/l2beat/pull/4520/commits/4be7c04321d675f4f274475d61e4dfad7e4d5118

File: packages/config/src/projects/other/da-beat/types/DaBridge.ts (hunk 1 above)
Comment (sdlyy) [reply packages/config/src/projects/other/da-beat/types/DaBridge.ts]: https://github.com/l2beat/l2beat/pull/4520/commits/5da67959e387e859cea288e24e632a2f4d9c1dca

File: packages/config/src/projects/other/da-beat/types/DaProjectReference.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { ProjectId } from '@l2beat/shared-pure'
+
+export type DaProjectReference = {
+  id: ProjectId
+  display: string
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/config/src/projects/other/da-beat/types/DaProjectReference.ts]: https://github.com/l2beat/l2beat/pull/4520/commits/5da67959e387e859cea288e24e632a2f4d9c1dca

**Review by torztomasz — APPROVED** [review]

**Pull request merged**